### PR TITLE
Allow controllers without a related model (rebased; updated)

### PR DESCRIPTION
**Pull request by dmitry-solomadin**
This is simply a rebase and a small documentation fix to the work done in this pull request https://github.com/thoughtbot/administrate/pull/1298

I've checked that it works correctly on a production application that used `custom-dashboard` branch previously

### CODE REVIEW

**Comment by sedubois** [comment]
@dmitry-solomadin could you fix the hound comments?

**Comment by dmitry-solomadin** [comment]
@sedubois Done

**Comment by sedubois** [comment]
OK now it's just about waiting for comments from @pablobm or @nickcharlton 🙂

File: lib/administrate/custom_dashboard.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  class CustomDashboard
+    include Administrate
+
+    class << self
+      def resource_name(_opts)
+        named_resource.pluralize.titleize
+      end
+
+      def resource(resource_name)
+        define_singleton_method(:named_resource) { resource_name }
```
Comment (guivinicius) [inline lib/administrate/custom_dashboard.rb]: What do you think about inferring the resource_name from its class? 

```ruby
resource_name || self.name.remove('Dashboard')
```

File: docs/adding_controllers_without_related_model.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Adding Controllers Without Related Model
```
Comment (nickcharlton) [inline docs/adding_controllers_without_related_model.md]:
```suggestion
# Adding Controllers without a related Model
```

File: docs/adding_controllers_without_related_model.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  <p><b>Total Customers:</b> <%= @stats[:customer_count] %></h1>
+  <br>
+  <p><b>Total Orders:</b> <%= @stats[:order_count] %></h1>
+</div>
+```
+
+```ruby
+# app/dashboards/stat_dashboard.rb
+
+require "administrate/custom_dashboard"
```
Comment (nickcharlton) [inline docs/adding_controllers_without_related_model.md]:
```suggestion
# app/dashboards/stat_dashboard.rb
require "administrate/custom_dashboard"
```

File: docs/adding_controllers_without_related_model.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+class StatDashboard < Administrate::CustomDashboard
+  resource "Stats" # used by administrate in the views
+end
+```
+
+```ruby
+# config/routes.rb
+
+namespace :admin do
```
Comment (nickcharlton) [inline docs/adding_controllers_without_related_model.md]:
```suggestion
# config/routes.rb
namespace :admin do
```

File: lib/administrate/custom_dashboard.rb (hunk 1 above)
Comment (nickcharlton) [reply lib/administrate/custom_dashboard.rb]: I don't have a strong opinion on this; can you see any downsides of taking this approach compared to what we're doing?

File: spec/example_app/app/views/layouts/docs.html.erb [hunk 5]
```diff
@@ -28,6 +28,7 @@
         <li><a href="/customizing_page_views">Customizing Page Views</a></li>
         <li><a href="/customizing_attribute_partials">Customizing Attribute Partials</a></li>
         <li><a href="/adding_custom_field_types">Adding Custom Field Types</a></li>
+        <li><a href="/adding_controllers_without_related_model">Adding Controllers Without Related Model</a></li>
```
Comment (nickcharlton) [inline spec/example_app/app/views/layouts/docs.html.erb]:
```suggestion
<li><a href="/adding_controllers_without_related_model">Adding Controllers without a related Model</a></li>
```

File: lib/administrate/custom_dashboard.rb (hunk 1 above)
Comment (dmitry-solomadin) [reply lib/administrate/custom_dashboard.rb]: Just in the sake of time - since we want to merge it in tomorrow - let's skip this change

**Comment by dmitry-solomadin** [comment]
@nickcharlton rebased from master and applied your comments - ty

**Pull request merged**